cfme/tests/infrastructure/test_host_cpu_resources.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from utils import testgen
+from cfme.web_ui import Quadicon, toolbar
+from cfme.infrastructure.host import Host
+from cfme.web_ui import InfoBlock
+from cfme.fixtures import pytest_selenium as sel
+from cfme.configure.tasks import is_host_analysis_finished
+from utils.wait import wait_for
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = \
+        testgen.infra_providers(metafunc, required_fields=["ssh_credentials"])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist,
+                        scope="module")
+
+
+def get_integer_value(x):
+    return int(x.split(' ')[0])
```

Comment:
Just a query on what gets returned here. I thought the ```InfoBlock``` widget used in the ```get_detail```, when run with ```.text()``` returned just the textual string. In which case just an ```int()``` casting should be good enough where you are using it, and we could drop that function. On the other hand if you are getting back MORE that just the text from the InfoBlock we either need another helper function in InfoBlock, or we need to fix the original method, perhaps adding a ```.integer()``` method.